### PR TITLE
[Issue #1686] add `created_at`, `updated_at`, and `deleted_at` columns to staging tables

### PR DESCRIPTION
**Pull request by jamesbursa**
## Summary
Part of #1686 (separate for easier review)

## Changes proposed
- Add `created_at`, `updated_at`, and `deleted_at` columns to staging tables
- Better logging of migrate queries

## Context for reviewers
These columns will provide helpful metadata for understanding or troubleshooting the load process.

The `created_at` and `updated_at` columns are set automatically by SQLAlchemy.

For `deleted_at`, I will update #1852 after this is merged.

## Additional information
Example from local testing:
Screenshot 2024-05-01 at 11 31 47

### CODE REVIEW

File: api/src/db/migrations/versions/2024_05_01_add_created_at_updated_at_and_deleted_.py [hunk 1]
```diff
@@ -0,0 +1,1168 @@
...
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table("topportunity_cfda", schema="legacy")
+    op.drop_table("t1", schema="api")
+    op.drop_table("tfundinstr_synopsis_hist", schema="legacy")
+    op.drop_table("tfundactcat_forecast_hist", schema="legacy")
+    op.drop_table("t2", schema="api")
+    op.drop_table("tfundinstr_synopsis", schema="legacy")
```
Comment (chouinar) [inline api/src/db/migrations/versions/2024_05_01_add_created_at_updated_at_and_deleted_.py:26]: You've got a lot going on in this migration that isn't the 3 timestamps you added, something is off here.

Comment (chouinar) [reply api/src/db/migrations/versions/2024_05_01_add_created_at_updated_at_and_deleted_.py:26]: Looks like its dropping the legacy foreign tables - I think we want to modify env.py to ignore anything in that schema for alembic

Comment (jamesbursa) [reply api/src/db/migrations/versions/2024_05_01_add_created_at_updated_at_and_deleted_.py]: Thanks, fixed

File: api/src/db/migrations/run.py [hunk 2]
```diff
@@ -30,13 +33,49 @@ def up(revision: str = "head") -> None:
...
 
 def downall(revision: str = "base") -> None:
+    enable_query_logging()
     command.downgrade(alembic_cfg, revision)
 
 
+def enable_query_logging() -> None:
+    """Log each migration query as it happens along with timing.
+
+    Based on the example at https://docs.sqlalchemy.org/en/13/faq/performance.html#query-profiling
```
Comment (chouinar) [inline api/src/db/migrations/run.py:48]: Current docs:
```suggestion
    Based on the example at https://docs.sqlalchemy.org/en/20/faq/performance.html#query-profiling
```

File: api/src/db/migrations/run.py [hunk 3]
```diff
@@ -30,13 +33,49 @@ def up(revision: str = "head") -> None:
...
+    enable_query_logging()
     command.downgrade(alembic_cfg, revision)
 
 
 def downall(revision: str = "base") -> None:
+    enable_query_logging()
     command.downgrade(alembic_cfg, revision)
 
 
+def enable_query_logging() -> None:
```
Comment (chouinar) [inline api/src/db/migrations/run.py:45]: Testing this, it ends up being very verbose. Is there a way we could make this configurable? Fine if its a follow-up, but locally I want to see just the alembic output, not this so much and now the logs are 3x as long

**Review by chouinar — APPROVED** [review]

**Review by chouinar — APPROVED** [review]

**Pull request merged**